### ISSUE TITLE
Frequency of upload progress decreased after 1.6.8

### ISSUE THREAD

**Issue by gepbird** [axios/axios#6526]
### Describe the bug

When uploading a file with axios 1.7.0 and above with an `onUploadProgress` callback provided, the callback will get called ~2.5 times per second compared to ~5 (or even ~15 on a close server) times per second in axios 1.6.8.

### To Reproduce

Go to https://codesandbox.io/p/sandbox/axios-slow-progress-1-7-3-3282wk, open console, click on upload, after waiting a bit you should see something like `Finished uploading file, axios made 2.28 progress reports per second` in the console.

### Code snippet

```js
// create a 50 MB file
  const size = 50 * 1024 * 1024;
  const chunkSize = 1024;
  const uint8 = new Uint8Array(chunkSize);
  let parts = [];
  for (let i = 0; i < size; i += chunkSize) {
    parts.push(uint8);
  }
  const blob = new Blob(parts);
  const file = new File([blob], "filename");

  // upload the file and count how many progress events did we get
  let count = 0;
  const startTime = Date.now();
  const formData = new FormData();
  formData.append("file", file);
  axios.post("https://httpbin.org/post", formData, {
    onUploadProgress: (e) => {
      console.log(`progress: ${e.progress}, count: ${count}`);
      count++;
      // NOTE: from 1.7.0 progress doesn't reach exactly 1, this is fixed in 1.7.3
      if (e.progress === 1) {
        const elapsedSecs = (Date.now() - startTime) / 1000;
        const reportsPerSec = Math.round((count / elapsedSecs) * 100) / 100;
        console.log(
          `Finished uploading file, axios made ${reportsPerSec} progress reports per second`
        );
      }
    },
  });
```

### Expected behavior

Faster upload speed that was before 1.7.0, to check it you can run the previous code with 1.6.8 or use https://codesandbox.io/p/sandbox/axios-fast-progress-1-6-8-rw7vdd?workspaceId=0994da35-60c6-4b2e-af43-fa484a57a293. You should see something similar to `Finished uploading file, axios made 6.2 progress reports per second`.

### Axios Version

1.7.3

### Adapter Version

default

### Browser

Firefox

### Browser Version

128.0b3

### Node.js Version

20.15.1

### OS

NixOS

### Additional Library Versions

```bash
-
```

### Additional context/Screenshots

```bash
Same as https://github.com/axios/axios/issues/6417#issuecomment-2143959421.
```

**Comment by gepbird** [comment]
bisected: a3ff99b59d8ec2ab5dd049e68c043617a4072e42 is the first bad commit

**Comment by gepbird** [comment]
@DigitalBrainJS is there a way to turn off the throttling introduced in #6371?

**Comment by DigitalBrainJS** [comment]
@gepbird No, although adding such an option is trivial. Are there any reasons why we need it?
I can’t find the related github issue, but someone asked to limit the firing of progress events, since this leads to frequent changes in the UI state and, accordingly, redrawing of components.
That sounds reasonable as smooth progress bar raising can be achieved with simple CSS animation and triggering an event every 20ms is not necessary.
But if the community wants to see an option that regulates or completely disables throttling for xhr, then I see no problem adding it.

**Comment by gepbird** [comment]
@DigitalBrainJS thanks for your quick response!

The reason why disabling throttling would be useful is the user could see a more up-to-date progress bar. Yes, it is only a few milliseconds more recent information, but in my opinion it improves the user experience by a noticeable amount. On the following video I used axios 1.7.3, top progress bar had a 0.2s transition (it was smooth on 1.6.8), bottom progress bar had a 0.5s transition:

https://github.com/user-attachments/assets/267ce0be-65a8-44b3-a82c-a7438b8fcfb6

The CSS for smoothing the progress bar: `transition: width ease-out 0.2s, background-color ease-out 0.2s;`.

I'd love to see an option where the throttling can be turned off, but I understand if it's low priority since turned out this is not a bug in axios.

**Comment by jasonsaayman** [comment]
great lets add this

**Comment by jasonsaayman** [comment]
Thanks for investigating this. The reduced event frequency is due to intentional throttling added to avoid excessive UI updates. The remaining request is for a way to configure or disable that throttling, which is tracked separately in #6378.

I am closing this as a duplicate of #6378.
